pandas/core/base.py

```diff
@@ -470,9 +470,12 @@ def is_any_frame() -> bool:
             try:
                 result = DataFrame(result)
             except ValueError:
-
                 # we have a dict of scalars
-                result = Series(result, name=getattr(self, "name", None))
+
+                # GH 36212 use name only if self is a series
+                name = getattr(self, "name", None) if (self.ndim == 1) else None
```

Comment:
No need to use getattr when the string is known. This can be simplified to `name = self.name if self.ndim == 1 else None`.

Reply:
Series are guaranteed to have a name attribute.